engine/src/main/java/org/terasology/world/block/regions/BlockRegionSystem.java

```diff
@@ -38,10 +43,25 @@
     @In
     private BlockManager blockManager;
 
-    @ReceiveEvent
+    // trivial priority so that all other logic can happen to the region before erasing the blocks in the region
+    @ReceiveEvent(priority = EventPriority.PRIORITY_TRIVIAL)
     public void onDestroyed(DoDestroyEvent event, EntityRef entity, BlockRegionComponent blockRegion) {
         for (Vector3i blockPosition : blockRegion.region) {
             worldProvider.setBlock(blockPosition, blockManager.getBlock(BlockManager.AIR_ID));
         }
     }
+
+    @ReceiveEvent
+    public void onActivatedActAsBlockAddHealth(OnActivatedComponent event, EntityRef entity, BlockRegionComponent blockRegion, ActAsBlockComponent actAsBlock) {
```

Comment:
I don't really agree with this. If the ActAsBlock entity should have health, the entity should just be given a Health component I feel. Health is too tied to blocks at the moment as is.